packages/redux/src/entities/selectors/entity.ts

```diff
@@ -42,36 +40,27 @@ export const getEntities = <
 /**
  * Gets a specific entity of the given schema name, by id.
  *
- * @memberof module:entities/selectors
- *
- * @param {object} state - Application state.
- * @param {string} name - Entity/schema name.
- * @param {string|number} id - Entity identifier.
+ * @param state - Application state.
+ * @param name - Entity/schema name.
+ * @param id - Entity identifier.
  *
- * @returns {object|undefined} - The specific entity of the given id, undefined
+ * @returns - The specific entity of the given id, undefined
  * if it isn't found.
  */
-export const getEntityById = <
-  SchemaName extends keyof NonNullable<StateEntities>,
-  EntityId extends keyof NonNullable<NonNullable<StateEntities>[SchemaName]>,
->(
+export const getEntityById = <T>(
   state: StoreState,
   name: SchemaName,
   id: EntityId,
-): NonNullable<StateEntities>[SchemaName][EntityId] | undefined =>
-  getEntities(state, name)?.[id];
+): T => getEntities(state, name)?.[id];
 
 /**
  * Returns a specific entity by given name and id.
  *
- * @function getEntity
- * @memberof module:entities/selectors
- *
- * @param   {object} state - Aplication state.
- * @param   {string} name  - Entity name.
- * @param   {string|number} id    - Entity identifier.
+ * @param state - Aplication state.
+ * @param name  - Entity name.
+ * @param id    - Entity identifier.
  *
- * @returns {object} The entity.
+ * @returns The entity.
```

Comment:
Probably out of scope but, having a `getEntities` and a `getEntityById`, do you think that our `getEntity` still makes sense?

If I recall correctly, `getEntity` was the previous selector we used, and it behaves somewhat unexpected. I'm just asking this because the documentation and the goal is very similar to what the other selectors do, but with no side effects. Food for thought 🙂

Reply:
I've checked and there's still a few files using `getEntity`, although the descriprtion for both functions looks the same, I noticed that `getEntity` accepts a null id. And some of the implementations are not sending the id.

But, since all those implementations are done on my area (account), I can check with my team and see if we can change that and remove that function :)

Reply:
Ok, just checked. So, what the function does is return the entity we want, without an id, and we (account) have a few use cases for that. This is one of them for example:

```export const getMerchants = (state: StoreState) => getEntity(state, 'merchants');```

For those cases like the one you mentioned in the below comment, the one where we are passing an Id, I will change it to use the "new" function. But for the rest of the cases I have to keep using `getEntity`.

Reply:
Ok, we can let it be for now, but I strongly recommend opening an issue with this, because it makes no sense to maintain 3 functions, where one of them is a mix of the other 2 (those 2 who were born to solve the problem that the 3rd has) 🧐 

I don't remember where we had it reported, but in a nutshell, a `getEntity` might return an entity (which is correct) or return a bunch of entities, if the id of the one provided isn't found (which is a side-effect, actually, not the intended) - for example, for `getEntity(state, 'merchants', 123)`, if the merchant `123` doesn't exist, it'll return all merchant entities.

Reply:
Yes, if it is a quick change I'll update on this PR, otherwise i'll create a task to be done in our team.